.github/workflows/license-check.yml

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Check License
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.event.number || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      MVN_COMMON_OPTIONS: -U -B --no-transfer-progress
+      MVN_BUILD_OUTPUT_FILE: "/tmp/mvn_build_output.out"
+      MVN_VALIDATION_DIR: "/tmp/fluss-validation-deployment"
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: 8
+          distribution: 'temurin'
+      - name: Build
+        run: |
+          set -o pipefail
+
+          ./mvnw clean deploy ${{ env.MVN_COMMON_OPTIONS }} -DskipTests \
+            -DaltDeploymentRepository=validation_repository::default::file:${{ env.MVN_VALIDATION_DIR }} \
+            | tee ${{ env.MVN_BUILD_OUTPUT_FILE }}
+
+      - name: Check License
+        run: |
+          ./mvnw ${{ env.MVN_COMMON_OPTIONS }} exec:java@check-license -N \
```

Comment:
Why not use `org.apache.rat:apache-rat-plugin:check` to check license?

Reply:
They checks different things. RAT plugin checks license header of Fluss source code. The `check-license` tool checks the necessary NOTICE files (and bundled dependencies in the NOTICE files).